tests/__tests__/api/issue_map/patchIssueMap.js

```diff
@@ -0,0 +1,212 @@
+      designers: ["invalid@email.com"],
+      photographers: ["ethan@ethan.com"],
+      section: validSectionName,
+      categories: "biology",
+    };
+    const response = await request(app)
+      .patch(`/issue-map/add-and-create-article`)
+      .set("Cookie", [`token=${tokens["ethan@ethan.com"]}`])
+      .send(requestBody);
+    showLog && console.log(response);
+    expect(response.status).toBe(404);
+    const errorMessage = JSON.parse(response.text).error;
+    expect(errorMessage).toStrictEqual("Invalid request body.");
+  });
+
+  test("Invalid article addition due to invalid photographers email", async () => {
+    const requestBody = {
+      articleSlug: validArticleSlug,
+      issueNumber: 1,
+      pageLength: 3,
+      authors: ["ethan@ethan.com"],
+      editors: ["ethan@ethan.com"],
+      designers: ["ethan@ethan.com"],
+      photographers: ["invalid@email.com"],
+      section: validSectionName,
+      categories: "biology",
+    };
+    const response = await request(app)
+      .patch(`/issue-map/add-and-create-article`)
+      .set("Cookie", [`token=${tokens["ethan@ethan.com"]}`])
+      .send(requestBody);
+    showLog && console.log(response);
+    expect(response.status).toBe(404);
+    const errorMessage = JSON.parse(response.text).error;
+    expect(errorMessage).toStrictEqual("Invalid request body.");
+  });
+
+  test("Invalid article addition due to article slug already existed", async () => {
+    const requestBody = {
+      articleSlug: validArticleSlug,
+      issueNumber: 1,
+      pageLength: 3,
+      authors: ["ethan@ethan.com"],
+      editors: ["ethan@ethan.com"],
+      designers: ["ethan@ethan.com"],
+      photographers: ["ethan@ethan.com"],
+      section: validSectionName,
+      categories: "biology",
+    };
+    const response = await request(app)
+      .patch(`/issue-map/add-and-create-article`)
+      .set("Cookie", [`token=${tokens["ethan@ethan.com"]}`])
+      .send(requestBody);
+    showLog && console.log(response);
+    expect(response.status).toBe(404);
+    const errorMessage = JSON.parse(response.text).error;
+    expect(errorMessage).toStrictEqual("Invalid request body.");
+  });
+
+  test("should create and add article to issue map successfully with missing authors/editors/designers/photographers", async () => {
+    const requestBody = {
+      articleSlug: "New article",
+      issueNumber: 2,
+      pageLength: 5,
+    };
+
+    const response = await request(app)
+      .patch(`/issue-map/add-and-create-article`)
+      .set("Cookie", [`token=${tokens["ethan@ethan.com"]}`])
+      .send(requestBody);
+
+    console.log(response.body);
+    expect(response.status).toBe(200);
+    const createdArticleID = response.body.articles.at(-1);
+    const article = await ArticlesAccessor.getArticle(createdArticleID);
+    console.log(article);
+    expect(article.issueNumber).toStrictEqual(requestBody.issueNumber);
+    expect(article.slug).toStrictEqual(requestBody.articleSlug);
```

Comment:
Consider adding a few more .toStrictEqual checks here to confirm that designStatus is set to Needs_Designer, photographyStatus is set to Needs_Photographer, etc